client/src/utils/optionalQueryString.ts

```diff
@@ -1,3 +1,7 @@
-export function optionalQueryString(value: string | undefined) {
-  return value ? String(value) : undefined;
+export function optionalQueryString(value: string[] | undefined) {
+  if (Array.isArray(value)) {
+    return String(value[0]);
+  }
+
+  return undefined;
```

Comment:
You don't need to explicitly return `undefined` as functions return `undefined` by default, please consider to return `null`/don't return at all